subsys/bluetooth/mesh/pb_gatt.c

```diff
@@ -118,16 +124,47 @@ static int link_accept(const struct prov_bearer_cb *cb, void *cb_data)
 	return 0;
 }
 
+static void pb_gatt_send_end(int err, void *cb_data)
+{
+	if (link.comp.cb) {
+		link.comp.cb(err, cb_data);
+	}
+}
+
 static int buf_send(struct net_buf_simple *buf, prov_bearer_send_complete_t cb,
 		    void *cb_data)
 {
+	int err;
+	struct net_buf *sbuf;
+
+	static const struct bt_mesh_send_cb pb_gatt_send_cb = {
+		.end = pb_gatt_send_end,
+	};
+
 	if (!link.conn) {
 		return -ENOTCONN;
 	}
 
+	sbuf = bt_mesh_adv_create(BT_MESH_ADV_PROV, 0, K_NO_WAIT);
+	if (!sbuf) {
+		return -ENOBUFS;
+	}
+
+	link.comp.cb = cb;
+	link.comp.cb_data = cb_data;
+
+	BT_MESH_ADV(sbuf)->cb = &pb_gatt_send_cb;
+	BT_MESH_ADV(sbuf)->cb_data = cb_data;
+
+	net_buf_simple_clone(buf, &sbuf->b);
```

Comment:
It'd be really nice if we'd be able to avoid this. Since the `bt_mesh_pb_gatt_send` function is now specific to this behavior, perhaps it can get some other solution? It could for instance allow passing its own `notify_complete` callback.

Reply:
If we move send_end to net_buf destroy callback, we can't change this.

We also need to count segment notify package, so net_buf_ref also needed.

Reply:
There's no need to copy the payload to an advbuf though? We're splitting and passing the packet to the host inline in the proxy module, couldn't we just call notify_complete from its callback, and not pull in the advertiser buffer at all for pb gatt?

Reply:
Done, Please take a look.